packages/cli/src/util/client.ts

```diff
@@ -67,6 +73,25 @@ export const isJSONObject = (v: any): v is JSONObject => {
   return v && typeof v == 'object' && v.constructor === Object;
 };
 
+export function isOAuthAuth(
+  authConfig: AuthConfig
+): authConfig is OAuthAuthConfig {
+  return authConfig.type === 'oauth';
+}
+
+export function isValidAccessToken(authConfig: OAuthAuthConfig): boolean {
+  return 'token' in authConfig && (authConfig.expiresAt ?? 0) >= Date.now();
+}
+
+export function isValidRefreshToken(
+  authConfig: OAuthAuthConfig
+): authConfig is OAuthAuthConfig & { refreshToken: string } {
+  return (
+    'refreshToken' in authConfig &&
+    (authConfig.refreshTokenExpiresAt ?? 0) >= Date.now()
+  );
```

Comment:
IMO, we could simplify this to
```suggestion
  return 'refreshToken' in authConfig
```